### ISSUE TITLE
Misalignment of 2D annotations

### ISSUE THREAD

**Issue by zamponotiropita** [scenenn/scenenn#5]
I want to label the raw RGBD data with 2D annotations. The depth and RGB data align well. Whereas the annotations do not seem to align with the raw RGBD data.

Take a look at the ladder on the right:
![scenenn](https://user-images.githubusercontent.com/23146389/33267642-88a1b14c-d37a-11e7-9cb4-808483ab7899.png)

Could you point me to how I need to transform the annotations (or rgbd) in order to align well?

**Comment by zamponotiropita** [comment]
Looks like the labeling is done on the depth but not transformed on RGB using extrinsics:

![image](https://user-images.githubusercontent.com/23146389/33460930-a343e21e-d630-11e7-9b04-91f7070d9a7b.png)

Depth (darker region) or RGB:
![image](https://user-images.githubusercontent.com/23146389/33460945-b0692256-d630-11e7-97d4-9d891e3f4930.png)

The images are from 213/image/image03404.png

@songuke could you please comment on that? Thank you.

**Comment by songuke** [comment]
@zamponotiropita The annotation was done on the 3D mesh reconstructed from the RGB-D video, then reproject to 2D. Most of the time this works, but in some scenes the reconstruction is not perfect (some misalignments during 3D reconstruction), and this causes misalignments in reprojection to 2D too. 

We attempted to address this issue by doing some optimizations to realign the boundaries (see here: http://people.sutd.edu.sg/~saikit/projects/sceneNN/pdf/annotation_tvcg17.pdf), but obviously this is not suitable if you want to address every frame in the scene. I think you might want to pick a scene with good 2D alignment if this really affects your application. 

(Somehow I was not notified when issues are posted on Github, hence this late response. I will check my Github again. Sorry for that.)

**Comment by KengChiLiu** [comment]
@songuke  In 2D annotation link you provide, it is empty in depth and rgb folder ???
Does scenenn provide ready both aligned RGBD and semantic annotation in 2D ???

thank you

**Comment by songuke** [comment]
Yes, the depth and RGB data are provided in the compressed RGB-D video in
the ONI files. If you need it, please use the provided playback tool to
export.

For semantic annotation, we provide the annotated data in 3D mesh. If you
need 2D information, we provided the annotation tool that can be used to
reproject all the annotation to 2D. The 2D annotation will be aligned to
the color image in most cases. Only in a few rare scenes, the 3D
reconstruction is not quite good, and so the 2D-3D alignment might be off a
little bit.

To obtain exact alignment, one way we thought about is to reproject both
depth, color, and annotation using the camera poses. The disadvantage of
this approach is that the reprojected color can be slightly blurry compared
to raw color image (since the reprojected color image is from the 3D mesh
textures).

2018-04-25 12:11 GMT+09:00 Keng-Chi Liu <notifications@github.com>:
